adapters/huaweiads/huaweiads.go

```diff
@@ -533,11 +534,24 @@ func getNativeFormat(adslot30 *adslot30, openRTBImp *openrtb2.Imp) error {
 		return err
 	}
 
+	//popular size for native ads
+	popularSizes := []map[string]int64{{"w": 225, "h": 150}, {"w": 300, "h": 250}, {"w": 320, "h": 250}, {"w": 640, "h": 360}, {"w": 720, "h": 1280}, {"w": 1080, "h": 170}, {"w": 1080, "h": 430}, {"w": 1080, "h": 432}, {"w": 1080, "h": 504}, {"w": 1080, "h": 607}, {"w": 1080, "h": 1620}, {"w": 1200, "h": 627}, {"w": 1280, "h": 720}, {"w": 1312, "h": 768}, {"w": 1920, "h": 1080}}
+
 	// only compute the main image number, type = native1.ImageAssetTypeMain
 	var numMainImage = 0
 	var numVideo = 0
-	var width int64
-	var height int64
+	var formats = make([]format, 0)
+	var numFormat = 0
+	var detailedCreativeTypeList = make([]string, 0, 2)
+
+	//number of the requested image size
+	for _, asset := range nativePayload.Assets {
+		if asset.Img != nil {
+			if asset.Img.Type == native1.ImageAssetTypeMain {
+				numFormat++
+			}
+		}
+	}
```

Comment:
following are the usage of `numFormat`
```
if numFormat > 1 && asset.Img.H != 0 && asset.Img.W != 0 && asset.Img.WMin != 0 && asset.Img.HMin != 0 {
					formats = append(formats, format{asset.Img.W, asset.Img.H})
				}
if numFormat == 1 && asset.Img.H != 0 && asset.Img.W != 0 && asset.Img.WMin != 0 && asset.Img.HMin != 0 {
					result := filterPopularSizes(popularSizes, asset.Img.W, asset.Img.H, "ratio")
					for i := 0; i < len(result); i++ {
						formats = append(formats, format{result[i]["w"], result[i]["h"]})
					}
				}
if numFormat == 1 && asset.Img.H == 0 && asset.Img.W == 0 && asset.Img.WMin != 0 && asset.Img.HMin != 0 {
					result := filterPopularSizes(popularSizes, asset.Img.W, asset.Img.H, "range")
					for i := 0; i < len(result); i++ {
						formats = append(formats, format{result[i]["w"], result[i]["h"]})
					}
```

Based on  usage we can break loop when `numFormat` becomes greater than 1.  This will save some extra loop iterations

```
	for _, asset := range nativePayload.Assets {
		if numFormat > 1 {
                    break
                 }

               if asset.Img != nil {
			if asset.Img.Type == native1.ImageAssetTypeMain {
				numFormat++
			}
		}
	}
```